src/main/java/duke/Parser.java

```diff
@@ -0,0 +1,124 @@
+package duke;
+
+import java.time.format.DateTimeParseException;
+import java.util.HashMap;
+import java.util.Map;
+
+public class Parser {
+    private final Map<String, ThrowingBiFunction<TaskList, String, String>> functions = new HashMap<>() {
+        {
+            put("list", (tasks, input) -> list(tasks));
+            put("done", (tasks, input) -> markDone(tasks, input));
+            put("todo", (tasks, input) -> addToDo(tasks, input));
+            put("deadline", (tasks, input) -> addDeadline(tasks, input));
+            put("event", (tasks, input) -> addEvent(tasks, input));
+            put("delete", (tasks, input) -> deleteTask(tasks, input));
+            put("find", (tasks, input) -> findTasks(tasks, input));
+        }
+    };
+
+    /**
+     * Parse the user input string and execute the respective command given in the functions Map.
+     * @param tasks Current Tasklist being used by the program.
+     * @param input Input String for the command.
+     * @return String Result to be printed by the UI class.
+     */
+    public String dispatch(TaskList tasks, String input) {
+        String[] cmd = input.split(" ", 2);
```

Comment:
Again very minor but I think names representing a collection of objects have to be plural, so maybe "cmds"?